### PR TITLE
check if we are adding a TabItem to TabControlRegion

### LINKED ISSUE #6: TabControlRegion does not correctly check if the incoming​ content is a TabItem
_From @mauroservienti on May 30, 2015 9:21_

Starts from the assumption that the incoming content needs to be wrapped in a TabItem.

_Copied from original issue: RadicalFx/Radical#162_

### PR DESCRIPTION
**Pull request by nazarenomanco**
Fix #6 
@mauroservienti, please check in I understand well

### CODE REVIEW

File: src/net40/Radical.Windows.Presentation/Regions/Specialized/TabControlRegion.cs [hunk 1]
```diff
@@ -39,16 +39,27 @@ public TabControlRegion( string name )
...
+                    Content = view
+                };
 
-            this.Element.Items.Add(tabItem);
+                object header = this.TryGetHeader(view);
+                if (header != null)
+                {
+                    tabItem.Header = header;
+                }                
+                this.Element.Items.Add(tabItem);
```
Comment (mauroservienti) [inline src/net40/Radical.Windows.Presentation/Regions/Specialized/TabControlRegion.cs]: The add has to be outside of the if/else.

File: src/net40/Radical.Windows.Presentation/Regions/Specialized/TabControlRegion.cs [hunk 2]
```diff
@@ -39,16 +39,27 @@ public TabControlRegion( string name )
...
         protected override void OnAdd(DependencyObject view)
         {
-            var tabItem = new TabItem();
-            tabItem.Content = view;
+            if (view == null) throw new ArgumentNullException(nameof(view));
 
-            var header = this.TryGetHeader(view);
-            if (header != null)
+            TabItem tabItem;
+            if (view is TabItem)
```
Comment (mauroservienti) [inline src/net40/Radical.Windows.Presentation/Regions/Specialized/TabControlRegion.cs]: There is a simpler way

```
var tabItem = view as TabItem;
if(tabItem == null)
{
   // create the TabItem to wrap the incoming view
}
```

File: src/net40/Radical.Windows.Presentation/Regions/Specialized/TabControlRegion.cs [hunk 3]
```diff
@@ -39,16 +39,27 @@ public TabControlRegion( string name )
         /// <param name="view">The view.</param>
         protected override void OnAdd(DependencyObject view)
         {
-            var tabItem = new TabItem();
-            tabItem.Content = view;
+            if (view == null) throw new ArgumentNullException(nameof(view));
```
Comment (mauroservienti) [inline src/net40/Radical.Windows.Presentation/Regions/Specialized/TabControlRegion.cs:42]: Drop this check, it should be added here https://github.com/RadicalFx/Radical.Windows.Presentation/blob/develop/src/net40/Radical.Windows.Presentation/Regions/ElementsRegion.cs#L62

**Review by mauroservienti — CHANGES_REQUESTED** [review]

File: src/net40/Radical.Windows.Presentation/Regions/Specialized/TabControlRegion.cs (hunk 1 above)
Comment (nazarenomanco) [reply src/net40/Radical.Windows.Presentation/Regions/Specialized/TabControlRegion.cs]: Ops 😄

**Comment by nazarenomanco** [comment]
@mauroservienti done! But not moved "TabItem check" at the moment

File: src/net40/Radical.Windows.Presentation/Regions/Specialized/TabControlRegion.cs (hunk 3 above)
Comment (nazarenomanco) [reply src/net40/Radical.Windows.Presentation/Regions/Specialized/TabControlRegion.cs:42]: The check and also the creation of the tab item that then we pass to this code? 

			this.OnAdd( view );
			this.views.Add( view );
			this.OnAdded( view );

Comment (mauroservienti) [reply src/net40/Radical.Windows.Presentation/Regions/Specialized/TabControlRegion.cs:42]: The check only, but should not be in this PR. Just remove it.

**Comment by nazarenomanco** [comment]
@mauroservienti check removed

**Comment by mauroservienti** [comment]
force pushed a `fixup` to revert unwanted changes to the `ElementsRegion` class.

**Review by mauroservienti — APPROVED** [review]

**Pull request merged**